From 89d044a6bf08fa3fb89b4e2bc829bad904cab705 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Wed, 28 Nov 2012 14:36:25 +0200
Subject: [PORT FROM MAIN] Telephony: Fix error in importing SIM contacts

BZ: 64280

Number of ADN records is not matching with the
number of IAP records resulting in the error log
"Error: Improper ICC card: No IAP record for ADN,
continuing".

EFiap, EFemail are Type 1 files that contain as
many records as the reference/master file(EFADN,
EFADN1). Currently, mIapFileRecord and
mEmailFileRecord are equated to the single EFiap
and EFemail content. Since there can be more than
1 EFiap, EFemail file, 1st file content will be
last when the 2nd file is loaded.

On loading of EFiap, EFemail, instead of equating
the contents, append the contents to the respective
list.

Change-Id: Ibbf37b46eac7683a04a131e8e37079ffe9f0bf5a
Orig-Change-Id: Ic66019448b15344773feda03e5ec4496c24675f7
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/gsm/UsimPhoneBookManager.java        |   11 +++++++----
 1 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/UsimPhoneBookManager.java b/src/java/com/android/internal/telephony/gsm/UsimPhoneBookManager.java
index 9a975e7..e2bad40 100755
--- a/src/java/com/android/internal/telephony/gsm/UsimPhoneBookManager.java
+++ b/src/java/com/android/internal/telephony/gsm/UsimPhoneBookManager.java
@@ -77,6 +77,9 @@ public class UsimPhoneBookManager extends Handler implements IccConstants {
     public UsimPhoneBookManager(IccFileHandler fh, AdnRecordCache cache) {
         mFh = fh;
         mPhoneBookRecords = new ArrayList<AdnRecord>();
+        mIapFileRecord = new ArrayList<byte[]>();
+        mEmailFileRecord = new ArrayList<byte[]>();
+
         mPbrFile = null;
         // We assume its present, after the first read this is updated.
         // So we don't have to read from UICC if its not present on subsequent reads.
@@ -86,8 +89,8 @@ public class UsimPhoneBookManager extends Handler implements IccConstants {
 
     public void reset() {
         mPhoneBookRecords.clear();
-        mIapFileRecord = null;
-        mEmailFileRecord = null;
+        mIapFileRecord.clear();
+        mEmailFileRecord.clear();
         mPbrFile = null;
         mIsPbrPresent = true;
         mRefreshCache = false;
@@ -358,7 +361,7 @@ public class UsimPhoneBookManager extends Handler implements IccConstants {
             log("Loading USIM IAP records done");
             ar = (AsyncResult) msg.obj;
             if (ar.exception == null) {
-                mIapFileRecord = ((ArrayList<byte[]>)ar.result);
+                mIapFileRecord.addAll((ArrayList<byte[]>)ar.result);
             }
             synchronized (mLock) {
                 mLock.notify();
@@ -368,7 +371,7 @@ public class UsimPhoneBookManager extends Handler implements IccConstants {
             log("Loading USIM Email records done");
             ar = (AsyncResult) msg.obj;
             if (ar.exception == null) {
-                mEmailFileRecord = ((ArrayList<byte[]>)ar.result);
+                mEmailFileRecord.addAll((ArrayList<byte[]>)ar.result);
             }
 
             synchronized (mLock) {
-- 
1.7.4.1

